cirq/circuits/circuit.py

```diff
@@ -2201,15 +2201,6 @@ def with_noise(self, noise: 'cirq.NOISE_MODEL_LIKE') -> 'cirq.Circuit':
         return c_noisy
 
 
-def _get_op_circuit(op: ops.Operation) -> Optional['cirq.FrozenCircuit']:
-    """Retrieves the circuit contained by an operation, if there is one."""
-    from cirq.circuits import CircuitOperation
-
-    while isinstance(op, ops.TaggedOperation):
-        op = op.sub_operation
```

Comment:
by reverting to op.untagged - don't we miss out in this recursion? `op.untagged` only goes down to a single layer of `suboperation` as far as I can tell.

Reply:
You're right, but I think this represents an issue in `TaggedOperation.untagged`: it claims to give a tag-free version of the operation, but might not if a nested `TaggedOperation` was created. (`TaggedOperation` has safeguards against nesting, but they're far from perfect.)

If you'd like, I can update `TaggedOperation.untagged` in this PR as well.

Reply:
I don't think there's a real use case for nested `TaggedOperations` - we explicitly guard against that behavior (though, as noted above, the protection is imperfect). My vote is in favor of either (1) only supported fully-recursive `untagged`, or (2) fully guarding against nesting (not sure if this is possible in Python).

Reply:
I don't think there should ever be a need to create nested `TaggedOperation`. If you use `op.with_tags` it will not create nested ops, and to me that seems like it should be sufficient. If we wanted to provide more assurance against people doing this accidentally we could modify `TaggedOperation.__init__` to flatten nested structures if you pass in a sub operation that is itself a `TaggedOperation`.

Reply:
oh nice, thanks for expanding on this - it makes sense!